provider/server/machine_networkinterface_attach_test.go

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package server_test
+
+import (
+	"time"
+
+	"github.com/digitalocean/go-libvirt"
+
+	iri "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
+	irimeta "github.com/ironcore-dev/ironcore/iri/apis/meta/v1alpha1"
+	libvirtutils "github.com/ironcore-dev/libvirt-provider/pkg/libvirt/utils"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"libvirt.org/go/libvirtxml"
+)
+
+var _ = Describe("AttachNetworkInterface", func() {
+	It("should attach a network interface to the machine", func(ctx SpecContext) {
+		By("creating a machine")
+		createResp, err := machineClient.CreateMachine(ctx, &iri.CreateMachineRequest{
+			Machine: &iri.Machine{
+				Metadata: &irimeta.ObjectMetadata{
+					Labels: map[string]string{
+						"foo": "bar",
+					},
+				},
+				Spec: &iri.MachineSpec{
+					Power: iri.Power_POWER_ON,
+					Class: machineClassx2medium,
+					Image: &iri.ImageSpec{
+						Image: osImage,
+					},
+				},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(createResp).NotTo(BeNil())
+
+		DeferCleanup(func(ctx SpecContext) {
+			Eventually(func() bool {
+				_, err := machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
+				Expect(err).ShouldNot(HaveOccurred())
+				_, err = libvirtConn.DomainLookupByUUID(libvirtutils.UUIDStringToBytes(createResp.Machine.Metadata.Id))
+				return libvirt.IsNotFound(err)
+			}).Should(BeTrue())
+		})
+
+		By("ensuring domain and domain XML is created for machine")
+		var domain libvirt.Domain
+		Eventually(func() error {
+			domain, err = libvirtConn.DomainLookupByUUID(libvirtutils.UUIDStringToBytes(createResp.Machine.Metadata.Id))
+			return err
+		}).WithTimeout(5 * time.Minute).WithPolling(5 * time.Second).Should(Succeed())
+		domainXMLData, err := libvirtConn.DomainGetXMLDesc(domain, 0)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(domainXMLData).NotTo(BeEmpty())
+
+		By("ensuring domain for machine is in running state")
+		Eventually(func() libvirt.DomainState {
+			domainState, _, err := libvirtConn.DomainGetState(domain, 0)
+			Expect(err).NotTo(HaveOccurred())
+			return libvirt.DomainState(domainState)
+		}).Should(Equal(libvirt.DomainRunning))
+
+		By("attaching network interface to the machine")
+		attachNetworkResp, err := machineClient.AttachNetworkInterface(ctx, &iri.AttachNetworkInterfaceRequest{
+			MachineId: createResp.Machine.Metadata.Id,
+			NetworkInterface: &iri.NetworkInterface{
+				Name: "nic-1",
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(attachNetworkResp).NotTo(BeNil())
+
+		By("ensuring network interface attached to the machine domain")
+		var interfaces []libvirtxml.DomainInterface
+		Eventually(func() int {
+			domainXMLData, err := libvirtConn.DomainGetXMLDesc(domain, 0)
+			Expect(err).NotTo(HaveOccurred())
+			domainXML := &libvirtxml.Domain{}
+			err = domainXML.Unmarshal(domainXMLData)
```

Comment:
Optional: If the only expected outcome is the absence of an error, we can streamline this by utilizing `Expect(domainXML.Unmarshal(domainXMLData)).Should(Succeed())`.

Please apply this modification consistently wherever else it's needed.

Reply:
Done, replaced wherever applicable